src/effects/Echo.cpp

```diff
@@ -163,3 +188,29 @@ EffectEcho::PopulateOrExchange(ShuttleGui & S, EffectSettingsAccess& access)
    S.EndMultiColumn();
    return nullptr;
 }
+
+
+bool EffectEcho::Validator::ValidateUI()
+{
+   mAccess.ModifySettings
+   (
+      [this](EffectSettings& settings)
+   {
+      // pass back the modified settings to the MessageBuffer
+      EffectEcho::GetSettings(settings) = mSettings;
+   }
+   );
+
+   return true;
+}
+
+
+bool EffectEcho::Validator::UpdateUI()
+{
+   // get the settings from the MessageBuffer and write them to our local copy
+   const auto& settings = mAccess.Get();
+   mSettings = GetSettings(settings);
```

Comment:
Again add this only at the last commit.

GetSettings should not be available until then.  The point where the echo settings move out of the effect and I to EffectSettings is only then.

Reply:
I commented it out + put a TODO comment "uncomment at final step"